src/analysis/normed_space/dual.lean

```diff
@@ -93,3 +119,192 @@ end
+
+lemma to_dual_map_isometry : isometry (@to_dual_map F _) :=
+add_monoid_hom.isometry_of_norm _ norm_to_dual_map_apply
+
+lemma to_dual_map_injective : function.injective (@to_dual_map F _) :=
+to_dual_map_isometry.injective
+
+@[simp] lemma ker_to_dual_map : (@to_dual_map F _).ker = ⊥ :=
+linear_map.ker_eq_bot.mpr to_dual_map_injective
+
+@[simp] lemma to_dual_map_eq_iff_eq {x y : F} : to_dual_map x = to_dual_map y ↔ x = y :=
+((linear_map.ker_eq_bot).mp (@ker_to_dual_map F _)).eq_iff
+
+variables [complete_space F]
+
+/--
+Fréchet-Riesz representation: any `ℓ` in the dual of a real Hilbert space `F` is of the form
+`λ u, ⟪y, u⟫` for some `y` in `F`.  See `inner_product_space.to_dual` for the continuous linear
+equivalence thus induced.
+-/
+-- TODO extend to `is_R_or_C` (requires a definition of conjugate linear maps)
+lemma range_to_dual_map : (@to_dual_map F _).range = ⊤ :=
+begin
+  apply linear_map.range_eq_top.mpr,
+  intros ℓ,
+  set Y := ker ℓ with hY,
+  by_cases htriv : Y = ⊤,
+  { have hℓ : ℓ = 0,
+    { have h' := linear_map.ker_eq_top.mp htriv,
+      rw [←coe_zero] at h',
+      apply coe_injective,
+      exact h' },
+    exact ⟨0, by simp [hℓ]⟩ },
+  { have Ycomplete := is_complete_ker ℓ,
+    rw [submodule.eq_top_iff_orthogonal_eq_bot Ycomplete, ←hY] at htriv,
+    change Y.orthogonal ≠ ⊥ at htriv,
+    rw [submodule.ne_bot_iff] at htriv,
+    obtain ⟨z : F, hz : z ∈ Y.orthogonal, z_ne_0 : z ≠ 0⟩ := htriv,
+    refine ⟨((ℓ z) / ⟪z, z⟫_ℝ) • z, _⟩,
+    ext x,
+    have h₁ : (ℓ z) • x - (ℓ x) • z ∈ Y,
+    { rw [mem_ker, map_sub, map_smul, map_smul, algebra.id.smul_eq_mul, algebra.id.smul_eq_mul,
+          mul_comm],
+      exact sub_self (ℓ x * ℓ z) },
+    have h₂ : (ℓ z) * ⟪z, x⟫_ℝ = (ℓ x) * ⟪z, z⟫_ℝ,
+    { have h₃ := calc
+        0    = ⟪z, (ℓ z) • x - (ℓ x) • z⟫_ℝ       :
+                  by { rw [(Y.mem_orthogonal' z).mp hz], exact h₁ }
+         ... = ⟪z, (ℓ z) • x⟫_ℝ - ⟪z, (ℓ x) • z⟫_ℝ  : by rw [inner_sub_right]
+         ... = (ℓ z) * ⟪z, x⟫_ℝ - (ℓ x) * ⟪z, z⟫_ℝ  : by simp [inner_smul_right],
+      exact sub_eq_zero.mp (eq.symm h₃) },
+    have h₄ := calc
+      ⟪((ℓ z) / ⟪z, z⟫_ℝ) • z, x⟫_ℝ = (ℓ z) / ⟪z, z⟫_ℝ * ⟪z, x⟫_ℝ
+            : by simp [inner_smul_left, conj_div, conj_conj]
+                            ... = (ℓ z) * ⟪z, x⟫_ℝ / ⟪z, z⟫_ℝ
+            : by rw [←div_mul_eq_mul_div]
+                            ... = (ℓ x) * ⟪z, z⟫_ℝ / ⟪z, z⟫_ℝ
+            : by rw [h₂]
+                            ... = ℓ x
+            : begin
+                have : ⟪z, z⟫_ℝ ≠ 0,
+                { change z = 0 → false at z_ne_0,
+                  rwa ←inner_self_eq_zero at z_ne_0 },
+                field_simp [this]
+              end,
+    exact h₄ }
+end
+
+/--
+Fréchet-Riesz representation: If `F` is a Hilbert space, the function that takes a vector in `F` to
+its dual is a continuous linear equivalence.  -/
+def to_dual : F ≃L[ℝ] (normed_space.dual ℝ F) :=
+continuous_linear_equiv.of_homothety
```

Comment:
Could you use `continuous_linear_equiv.of_isometry` instead? (You have introduced it precisely for this purpose :-)

Reply:
Indeed, but I think the code moved around a bit since then! But yes, it is still shorter this way.